benchto-driver/src/main/java/com/teradata/benchto/driver/execution/BenchmarkExecutionDriver.java

```diff
@@ -94,32 +94,38 @@ public BenchmarkExecutionResult execute(Benchmark benchmark, int benchmarkOrdina
 
     private BenchmarkExecutionResult executeBenchmark(Benchmark benchmark)
     {
+        BenchmarkExecutionResultBuilder resultBuilder = new BenchmarkExecutionResultBuilder(benchmark);
+        List<QueryExecutionResult> executions;
         try {
             executeQueries(benchmark, benchmark.getPrewarmRuns(), false);
 
             executionSynchronizer.awaitAfterBenchmarkExecutionAndBeforeResultReport(benchmark);
 
             statusReporter.reportBenchmarkStarted(benchmark);
 
-            BenchmarkExecutionResultBuilder resultBuilder = new BenchmarkExecutionResultBuilder(benchmark)
-                    .startTimer();
+            resultBuilder = resultBuilder.startTimer();
 
-            List<QueryExecutionResult> executions = executeQueries(benchmark, benchmark.getRuns(), true);
-
-            BenchmarkExecutionResult executionResult = resultBuilder
-                    .endTimer()
-                    .withExecutions(executions)
+            try {
+                executions = executeQueries(benchmark, benchmark.getRuns(), true);
+            }
+            finally {
+                resultBuilder = resultBuilder.endTimer();
+            }
+        }
+        catch (RuntimeException e) {
+            return resultBuilder
+                    .withUnexpectedException(e)
                     .build();
+        }
 
-            executionSynchronizer.awaitAfterBenchmarkExecutionAndBeforeResultReport(benchmark);
+        BenchmarkExecutionResult executionResult = resultBuilder
+                .withExecutions(executions)
+                .build();
 
-            statusReporter.reportBenchmarkFinished(executionResult);
+        executionSynchronizer.awaitAfterBenchmarkExecutionAndBeforeResultReport(benchmark);
```

Comment:
Even if it did, we should not report `failedBenchmarkResult` in such a case, and we should fail loudly instead.